### PR TITLE
Issue 19 - Adding Translate Object feature to Versions in the Hubs View

### PR DESCRIPTION
**Pull request by smorel**
**ISSUE**

https://github.com/petrbroz/vscode-forge-tools/issues/19

**DESCRIPTION**

Adding the "Translate Object" feature when right clicking on a version in the Hubs view.
This is a first action supported for issue number 19.

* The Hubs item types have been moved within a separate file in the interface folder so that we can use these types in model derivative commands or in the extension.

* The TranslateObject command definition has been updated in package so that it targets both "Objects" in data management and "Versions" in Hubs view.

* Updated the authentication scopes so our token has the right to trigger a translation.

* The translate Object command code now supports both Objects and Versions

* Refreshing the hubs view after TranslateObject command is executed so that we can start tracking the SVF manifest status and update progress, success or failures accordingly.

### CODE REVIEW

File: src/commands/authentication.ts [hunk 1]
```diff
@@ -3,11 +3,18 @@ import * as url from 'url';
...
 
-const DefaultScopes = ['viewables:read', 'bucket:read', 'bucket:create', 'data:read', 'data:create', 'data:write', 'code:all']; // Make this list configurable?
+const DefaultScopes = [
+    'viewables:read', 
+    'code:all', 
+    'bucket:create','bucket:read','bucket:update', 'bucket:delete', 
+    'data:read', 'data:write', 'data:create', 'data:search',
+    'account:read', 'account:write', 
+    'user:read', 'user:write',
+    'user-profile:read']; // Make this list configurable?
```
Comment (petrbroz) [inline src/commands/authentication.ts:13]: Good idea :+1:

File: package.json [hunk 2]
```diff
@@ -533,7 +533,7 @@
 				},
 				{
 					"command": "forge.translateObject",
-					"when": "view == forgeDataManagementView && viewItem == object",
+					"when": "viewItem == object || viewItem == version",
```
Comment (petrbroz) [inline package.json:536]: Could we keep the filtering by view? I understand that the command may now appear in two different views but I wouldn't solve that by simply removing the `view === ...` part. There could be other views with nodes of type `version` (for example, in the Design Automation tree), and we don't want the command to show up for those.

File: src/commands/model-derivative.ts [hunk 3]
```diff
@@ -25,7 +26,8 @@ function urnify(id: string): string {
 	return _urnify(id).replace('/', '_');
 }
 
-export async function translateObject(object: IObject | undefined, context: IContext) {
+//hi.IVersion
```
Comment (petrbroz) [inline src/commands/model-derivative.ts:29]: What's the purpose of this comment?

**Review by petrbroz — CHANGES_REQUESTED** [review]
Looks good, thanks Sebastien! Before we merge this in, let's address some of the minor comments, especially the `when` clause of the translate command.

File: src/commands/model-derivative.ts (hunk 3 above)
Comment (smorel) [reply src/commands/model-derivative.ts:29]: nothing good catch will remove :)

File: package.json (hunk 2 above)
Comment (smorel) [reply package.json:536]: I tried originally but I didn't succeed in having multiple and/or conditions. I can spend more time trying to find the way to express more complex when conditions.

File: src/commands/model-derivative.ts (hunk 3 above)
Comment (smorel) [reply src/commands/model-derivative.ts]: Ok found a way by having 2 command definitions with different when clauses.

**Comment by smorel** [comment]
@petrbroz  I addressed your comment let me know if you're good with the changes. cheers.

File: package.json (hunk 2 above)
Comment (smorel) [reply package.json]: Ok found a way by having 2 command definitions with different when clauses.

**Pull request merged**